### PR TITLE
[JENKINS-42717] - Prevent NPE when a non-existent Default View is specified in the global config

### PR DESCRIPTION
**Pull request by oleg-nenashev**
# Description

- [x] Prevent NPE in `GlobalDefaultViewConfiguration` when the specified view is missing
- [x] Annotate View management methods to let FindBugs notice the issue
- [x] Add a direct Unit test (WiP)

See [JENKINS-42717](https://issues.jenkins-ci.org/browse/JENKINS-42717).

### Changelog entries

Proposed changelog entries:

* Entry 1: JENKINS-42717]- Prevent NPE when a non-existent default view is specified in the Global Configuration
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Link to JIRA ticket in description, if appropriate
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc)

### CODE REVIEW

File: test/src/test/java/org/kohsuke/stapler/MockStaplerRequestBuilder.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+public class MockStaplerRequestBuilder{
+    
+    private final JenkinsRule r;
+    
+    public MockStaplerRequestBuilder(@Nonnull JenkinsRule r) {
+        this.r = r;
+    }
+    
+    public StaplerRequest build() throws AssertionError {
+        RequestImpl currentRequest = (RequestImpl) Stapler.getCurrentRequest();
```
Comment (oleg-nenashev) [inline test/src/test/java/org/kohsuke/stapler/MockStaplerRequestBuilder.java]: Likely the autotest will get null here. COpy-pasted it from ProgressiveHtml, will likely need a better mock to make the test passing

File: core/src/main/java/hudson/views/GlobalDefaultViewConfiguration.java [hunk 2]
```diff
@@ -41,7 +42,18 @@
     public boolean configure(StaplerRequest req, JSONObject json) throws FormException {
         // for compatibility reasons, the actual value is stored in Jenkins
         Jenkins j = Jenkins.getInstance();
-        j.setPrimaryView(json.has("primaryView") ? j.getView(json.getString("primaryView")) : j.getViews().iterator().next());
+        if (json.has("primaryView")) {
+            final String viewName = json.getString("primaryView");
+            final View newPrimaryView = j.getView(viewName);
+            if (newPrimaryView == null) {
+                throw new FormException("The specified primary view does not exist: " + viewName, "primaryView");
```
Comment (daniel-beck) [inline core/src/main/java/hudson/views/GlobalDefaultViewConfiguration.java]: Shown on the UI so should be i18n aware.

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/hudson/views/GlobalDefaultViewConfiguration.java (hunk 2 above)
Comment (oleg-nenashev) [inline core/src/main/java/hudson/views/GlobalDefaultViewConfiguration.java]: Uh, OK

**Comment by oleg-nenashev** [comment]
Fixed the mocking logic in the test

File: test/src/test/java/org/kohsuke/stapler/MockStaplerRequestBuilder.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        return this;
+    }
+    
+    public MockStaplerRequestBuilder withAncestor(AncestorImpl ancestor) {
+        this.ancestors.add(ancestor);
+        return this;
+    }
+       
+    public StaplerRequest build() throws AssertionError {        
+        HttpServletRequest rawRequest = Mockito.mock(HttpServletRequest.class);
```
Comment (oleg-nenashev) [inline test/src/test/java/org/kohsuke/stapler/MockStaplerRequestBuilder.java]: It was the most straightforward way to do that for now, will need a better impl if the method really presumes any request handling later

**Comment by oleg-nenashev** [comment]
@daniel-beck and @jenkinsci/code-reviewers ready for re-review

**Review by daniel-beck — APPROVED** [review]

**Pull request merged**